embedding-reports/host-the-report-engine-remotely/rest-service-storage/overview.md

```diff
@@ -29,6 +29,20 @@ The following implementations of the IStorage interface support the service to b
 * [DatabaseStorage](/api/Telerik.Reporting.Cache.Database.DatabaseStorage) - Use the [DatabaseStorage constructor](/api/Telerik.Reporting.Cache.Database.DatabaseStorage#Telerik_Reporting_Cache_Database_DatabaseStorage_#ctor) to create a storage instance. This storage option requires reference to __Telerik.Reporting.Cache.Database.dll__ that has dependencies on Telerik Data Access which can be checked in the version corresponding [Upgrade article]({%slug telerikreporting/upgrade/overview%}).
 
+## Cleaning the Reporting REST Service Storage
+
+The REST Service Storage keeps all the information about the registered clients/viewers. For example, the generated report documents, the images embedded in the reports, etc. Therefore, with each generated report the occupied storage space increases. To prevent the storage from growing unnecessarily, when a client expires, the Reporting Service triggers a cleanup routine deleting its assets.
+
+You may control the assets’ expiration using the [ClientSessionTimeout](/api/telerik.reporting.services.reportserviceconfiguration#Telerik_Reporting_Services_ReportServiceConfiguration_ClientSessionTimeout) and [ReportSharingTimeout](/api/telerik.reporting.services.reportserviceconfiguration#Telerik_Reporting_Services_ReportServiceConfiguration_ReportSharingTimeout) properties of the service configuration.
+
+When a viewer gets closed, there is no notification, and the service does not know about that. Therefore, when a viewer is inactive for a certain period determined by the `ClientSessionTimeout`, its resources are considered expired. Periodically (every 5 minutes), the REST Service performs cleaning of the expired data. The service triggers this logic only when awakened by a request from an active client/viewer. Hence, all inactive sessions will be deleted after some time determined by the `ClientSessionTimeout` and the time it takes to delete the expired data, provided the service is awake. You may prevent the expiration of an active viewer by setting the viewer's option `keepClientAlive` to `true` (default).
+
+Let's overview the relation between the clients' assets. Each client holds 'refreshes' to particular report 'documents'. These are the rendered report documents when previewing, exporting, or printing reports with particular parameter values and action states. Each report 'document' holds a reference to a report 'instance', which is the report definition instance. When the client expires, all storage key-value pairs associated with its _clientId_ are cleared along with all 'refreshes' that they hold. When there are no 'refreshes' to a 'document', it is considered expired and gets deleted along with the reference to the corresponding report 'instance'. The 'instance' gets removed when no 'documents' refer to it. Therefore, when a client expires, the 'instances' and 'documents' it references may remain.
```

Comment:
I think from here on you explain it in such detail that it will bring more confusion than clarity. Additionally, some sentences are pretty complex which also reduces clarity. I would explain it using another approach:
The assets cached when a report gets rendered are generated in hierarchical manner with each resource holding reference to its parent resource. To release a particular resource, the system counts the child resources that depend on this resource and if there are none, it gets released. Here are the resources from child to parent:
- client session - Kept for "ClientSessionTimeout" minutes (5 by default) after the last request from the originating client. The client session keeps the requested document refreshes alive.
- document refresh - A representation of report document rendered at particular moment in time. Contains the actual rendering assets like pages, images, docs (for the export format). Keeps the originating report instance alive
- report instance - this is the report definition with particular report parameter values, i.e., the report address.

...then explain that document refreshes can be reused setting the sharing timeout, etc. Hope this helps.